src/services/keystore.js

```diff
@@ -373,6 +366,100 @@ function updateWallet(
   return appendWallet(newWallets, newWallet)
 }
 
+function getPrivateKey(wallets: Wallets, walletId: string, password: string): string {
+  const wallet: Wallet = getWallet(wallets, walletId)
+
+  const {
+    encrypted,
+    passwordOptions,
+    type,
+    isReadOnly,
+  }: Wallet = wallet
+
+  checkNotReadOnly(isReadOnly)
+
+  if (!passwordOptions) {
+    throw new Error('Invalid wallet')
+  }
+
+  const {
+    salt,
+    scryptParams,
+    encryptionType,
+  }: PasswordOptions = passwordOptions
+
+  const dKey: Uint8Array = utils.deriveKeyFromPassword(password, salt, scryptParams)
+
+  if (type === MNEMONIC_TYPE) {
```

Comment:
how type here can be equal `MNEMONIC_TYPE`
I'm really confused with logic and checks.
https://github.com/jibrelnetwork/jwallet-web/pull/350/files#diff-1255dd3bf7cc56a6f5c0699df1fda04dR456 ?